src/Riok.Mapperly/Descriptors/SymbolAccessor.cs

```diff
@@ -519,4 +547,66 @@ private IEnumerable<IMappableMember> GetAllAccessibleMappableMembersCore(ITypeSy
             .Select(x => MappableMember.Create(this, x))
             .WhereNotNull();
     }
+
+    public INamedTypeSymbol? GetTypeByMetadataName(string targetTypeName, SyntaxNode? contextNode)
```

Comment:
I created a method that resolves the class by name within the given context. In the tests I've seen, there are more than 400 possibilities to choose from in that simple case. Although the result is cached, I'm doubting it is the best approach. Should I allow only the full path in case of a string?